service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ManageHearingsController.java

```diff
@@ -0,0 +1,178 @@
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = getCaseData(caseDetails);
+
+        //Only triggered for edit journey
+        if (EDIT_DRAFT.equals(caseData.getUseExistingHearing())) {
+            UUID hearingBookingId = getDynamicListValueCode(caseData.getHearingDateList(), mapper);
+
+            List<Element<HearingBooking>> futureHearings = hearingBookingService.getFutureHearings(
+                caseData.getHearingDetails());
+
+            caseDetails.getData().put(HEARING_DATE_LIST,
+                asDynamicList(futureHearings, hearingBookingId, hearing -> hearing.toLabel(DATE)));
+
+            HearingBooking hearingBooking = manageHearingsService.findHearingBooking(
+                hearingBookingId, caseData.getHearingDetails());
+
+            caseDetails.getData().putAll(manageHearingsService.populateHearingCaseFields(hearingBooking));
+
+            if (hearingBookingId.equals(caseData.getHearingDetails().get(0).getId())) {
+                caseDetails.getData().put(FIRST_HEARING_FLAG, "Yes");
+            }
+        }
+
+        return respond(caseDetails);
+    }
+
+    @PostMapping("/validate-hearing-dates/mid-event")
+    public AboutToStartOrSubmitCallbackResponse validateHearingDatesMidEvent(
+        @RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = getCaseData(caseDetails);
+
+        List<String> errors = validateGroupService.validateGroup(caseData, HearingDatesGroup.class);
+
+        return respond(caseDetails, errors);
+    }
+
+    @PostMapping("/about-to-submit")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = getCaseData(caseDetails);
+
+        HearingBooking hearingBooking = manageHearingsService.buildHearingBooking(caseData);
+
+        if (caseData.getSendNoticeOfHearing() != null && caseData.getSendNoticeOfHearing().equals("Yes")) {
+            manageHearingsService.addNoticeOfHearing(caseData, hearingBooking);
+        }
+
+        List<Element<HearingBooking>> hearingBookingElements;
+
+        // Editing previous hearing
+        if ((caseData.getUseExistingHearing() != null) && EDIT_DRAFT.equals(caseData.getUseExistingHearing())) {
+            UUID editedHearingId = getDynamicListValueCode(caseData.getHearingDateList(), mapper);
+
+            caseDetails.getData().put(SELECTED_HEARING_IDS, List.of(editedHearingId));
+
+            hearingBookingElements = manageHearingsService.updateEditedHearingEntry(
+                hearingBooking, editedHearingId, caseData.getHearingDetails());
+        } else {
+            hearingBookingElements = manageHearingsService.appendHearingBooking(
+                defaultIfNull(caseData.getHearingDetails(), List.of()), hearingBooking);
```

Comment:
I feel like the default getter in case data should return an empty list if it is null

Reply:
Will handle that in https://tools.hmcts.net/jira/browse/FPLA-2280 during refactor + cleanup